jobs/update-failed-distributions/jobs/statement_notification_job.py

```diff
@@ -0,0 +1,99 @@
+from pay_api.models.statement_recipients import StatementRecipients as StatementRecipientsModel
+from pay_api.services.oauth_service import OAuthService
+from pay_api.utils.enums import NotificationStatus, AuthHeaderType, ContentType
+from pay_api.utils.util import get_local_time
+from utils.auth import get_token
+
+ENV = Environment(loader=FileSystemLoader('.'), autoescape=True)
+
+
+class StatementNotificationJob:
+
+    @classmethod
+    def send_notifications(cls):
+        """Send Notifications.
+        Steps:
+        1. Get all statements with Notification status as PENDING
+        2. Start processing each statements
+        2. Set status as processing so that if there is one more job , it wont be fetched
+        3. Trigger mail
+        4. Check status and update back
+        """
+        statements_with_pending_notifications = StatementModel.find_all_statements_by_notification_status(
+            (NotificationStatus.PENDING.value,))
+        if statement_len := len(statements_with_pending_notifications) < 1:
+            current_app.logger.info('No Statements with Pending notifications Found!')
+            return
+        else:
+            current_app.logger.info(f'{statement_len} Statements with Pending notifications Found!')
+        token = get_token()
+
+        params = {
+            'logo_url': f"{current_app.config.get('AUTH_WEB_URL')}/{current_app.config.get('REGISTRIES_LOGO_IMAGE_NAME')}",
+            'url': f"{current_app.config.get('AUTH_WEB_URL')}/{current_app.config.get('AUTH_WEB_STATEMENT_URL')}"
+        }
+        template = ENV.get_template('statement_notification.html')
+        for statement in statements_with_pending_notifications:
+            statement.notification_status_code = NotificationStatus.PROCESSING.value
+            statement.notification_date = get_local_time(datetime.now())
+            statement.commit()
+            payment_account = PaymentAccountModel.find_by_id(statement.payment_account_id)
+            recipients = StatementRecipientsModel.find_all_recipients_for_payment_id(statement.payment_account_id)
+            to_emails = ','.join([str(recipient.email) for recipient in recipients])
+            params['org_name'] = payment_account.auth_account_name
+            params['frequency'] = statement.frequency
+            params.update({'url': params['url'].replace('orgId', payment_account.auth_account_id)})
+            notify_response = cls.send_email(token, to_emails, template.render(params))
+            if not notify_response:
+                current_app.logger.error('<notification failed')
+                statement.notification_status_code = NotificationStatus.FAILED.value
+                statement.notification_date = get_local_time(datetime.now())
+                statement.commit()
+            statement.notification_status_code = NotificationStatus.SUCCESS.value
+            statement.notification_date = get_local_time(datetime.now())
```

Comment:
Doesnt need to put local time, all the DB values needs to go in as UTC

Reply:
yeah..missed it..

Reply:
done